activerecord/lib/active_record/associations/builder/association.rb

```diff
@@ -30,6 +30,11 @@ def self.build(model, name, scope, options, &block)
       end
 
       reflection = create_reflection(model, name, scope, options, &block)
+
+      if reflection.belongs_to? && model.ignored_columns.include?(reflection.foreign_key.to_s)
+        raise ArgumentError, "You tried to define a belongs_to association for #{name}, but the column #{reflection.foreign_key} is ignored. You can remove it from the ignored_columns list if you want to use it."
+      end
+
```

Comment:
Sorry, I meant adding it to `Builder::BelongsTo`:
https://github.com/rails/rails/blob/main/activerecord/lib/active_record/associations/builder/belongs_to.rb

```ruby
module ActiveRecord::Associations::Builder # :nodoc:
  class BelongsTo < SingularAssociation # :nodoc:
    ...
    def self.build(model, name, scope, options, &block)
      if model.ignored_columns.include?(reflection.foreign_key.to_s)
        raise ArgumentError, "You tried to define a belongs_to association for #{name}, but the column #{reflection.foreign_key} is ignored. You can remove it from the ignored_columns list if you want to use it."
      end
      super
    end
```

Reply:
I thought about it also but we need to call the super first to get the reflection first, and reflections runs all the checks 

```
      define_accessors(model, reflection)
      define_callbacks(model, reflection)
      define_validations(model, reflection)
      define_change_tracking_methods(model, reflection)
 ```
 
 and I thought an yearly return might be useful.

Reply:
I would like to propose a validate_reflection interface which can be overriden by any child class and and leave an option to validate a reflection if needed.

```
    def self.validate_reflection!(model, reflection)
      # noop
    end
```

Reply:
@p8 I proposed a suggestion, feel free to review and let me know if that is okey.